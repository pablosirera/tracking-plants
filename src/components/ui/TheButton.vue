<template>
  <button type="submit" :class="buttonClasses" :disabled="isDisabled">
    <slot />
  </button>
</template>

<script>
export default {
  name: 'TheButton',
  props: {
    theme: {
      type: String,
      default: 'primary',
      validate: value => ['primary'].indexOf(value) !== -1
    },
    size: {
      type: String,
      default: 'medium',
      validate: value => ['small', 'medium', 'large'].indexOf(value) !== -1
    },
    isDisabled: {
      type: Boolean,
      default: false
    }
  },
  computed: {
    buttonClasses() {
      const commonClasses =
        'border py-2 px-6 rounded-full font-bold focus:outline-none'
      const primaryClasses = `${commonClasses} bg-green-600 text-white`

      const mediumClasses = 'w-3/5'
      const largeClasses = 'w-full'

      return {
        [primaryClasses]: this.theme === 'primary',
        [mediumClasses]: this.size === 'medium',
        [largeClasses]: this.size === 'large',
        'opacity-50 cursor-default': this.isDisabled
      }
    }
  }
}
</script>
