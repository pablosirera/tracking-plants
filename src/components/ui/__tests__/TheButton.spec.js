import { shallowMount } from '@vue/test-utils'
import TheButton from '../TheButton'

describe('Global Components -> Ui -> TheButton', () => {
  const build = propsData => {
    const options = propsData ? { propsData } : null
    const wrapper = shallowMount(TheButton, options)

    return {
      wrapper
    }
  }

  it('should match snapshot', () => {
    const { wrapper } = build()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should add the primary theme classes when theme prop is primary', () => {
    const { wrapper } = build({ theme: 'primary' })
    const button = wrapper.find('button')
    const buttonClasses = button.classes()

    expect(buttonClasses).toContain('bg-green-600')
    expect(buttonClasses).toContain('text-white')
  })

  it('should add the medium classes when size is medium', () => {
    const { wrapper } = build({ size: 'medium' })
    const button = wrapper.find('button')
    const buttonClasses = button.classes()

    expect(buttonClasses).toContain('w-3/5')
  })

  it('should add the large classes when size is large', () => {
    const { wrapper } = build({ size: 'large' })
    const button = wrapper.find('button')
    const buttonClasses = button.classes()

    expect(buttonClasses).toContain('w-full')
  })
})
