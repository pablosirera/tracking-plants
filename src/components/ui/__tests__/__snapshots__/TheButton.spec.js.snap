// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Global Components -> Ui -> TheButton should match snapshot 1`] = `
<button
  class="border py-2 px-6 rounded-full font-bold focus:outline-none bg-green-600 text-white w-3/5"
  type="submit"
/>
`;
